test/functional/inputs/http/http_input_test.go

```diff
@@ -0,0 +1,137 @@
+				Hostname:                 `*`,
+				LogType:                  `audit`,
+				Level:                    `default`,
+				AuditID:                  `16dcc977-a39e-467b-af44-921416b7800b`,
+				Stage:                    `RequestReceived`,
+				RequestURI:               `/apis/rbac.authorization.k8s.io/v1/clusterrolebindings?limit=500\u0026resourceVersion=0`,
+				Timestamp:                tsF(`2022-08-17T20:27:20.570375Z`),
+				RequestReceivedTimestamp: tsF(`2023-08-24T21:37:38.842649Z`),
+				ViaqMsgID:                "*",
+				PipelineMetadata:         functional.TemplateForAnyPipelineMetadata,
+				OpenshiftLabels: types.OpenshiftMeta{
+					ClusterID: "*",
+					Sequence:  types.NewOptionalInt(""),
+				},
+			},
+			K8SAuditLevel: "default",
+		}
+	)
+
+	BeforeEach(func() {
+		Expect(testfw.LogCollectionType).To(Equal(logging.LogCollectionTypeVector))
+		framework = functional.NewCollectorFunctionalFrameworkUsingCollector(logging.LogCollectionTypeVector)
+		functional.NewClusterLogForwarderBuilder(framework.Forwarder).
+			FromInputWithVisitor("http-source",
+				func(spec *logging.InputSpec) {
+					spec.Source = &logging.SourceSpec{
+						HTTP: &logging.HTTPSource{
+							Name: `http-source-spec-name`,
+							Port: v1.ServicePort{
+								Port: servicePort,
+							},
+							Format:  logging.FormatK8S,
+							LogType: `audit`,
+						},
+					}
+				}).ToHttpOutput()
+	})
+
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	Context("When sending an audit log record to an HTTP input", func() {
+		It("should be able to round trip it unharmed", func() {
+			Expect(framework.DeployWithVisitor(
+				func(b *runtime.PodBuilder) error {
+					return framework.AddVectorHttpOutput(b, framework.Forwarder.Spec.Outputs[0])
+				}),
+			).To(BeNil())
+
+			_, err := framework.RunCommand(constants.CollectorName, "curl", "-ksv", fmt.Sprintf("http://localhost:%d", servicePort), "-d",
+				func() string { buf, _ := json.Marshal(auditRecord); return string(buf) }(),
+			)
+			Expect(err).To(BeNil(), "Expected no errors reading the logs")
+
+			raw, err := framework.ReadFileFrom("http", functional.ApplicationLogFile)
+			Expect(err).To(BeNil(), "Expected no errors reading the logs")
+			var logs []types.K8sAuditLog
+			err = types.ParseLogsFrom(utils.ToJsonLogs([]string{raw}), &logs, false)
+			Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+			fmt.Fprintln(GinkgoWriter, raw)
+			outputTestLog := logs[0]
+			Expect(outputTestLog).To(FitLogFormatTemplate(auditRecord))
+		})
+	})
+
+	Context("When sending an array of audit log records to an HTTP input", func() {
+		It("should be able to round trip them unharmed", func() {
+			Expect(framework.DeployWithVisitor(
+				func(b *runtime.PodBuilder) error {
+					return framework.AddVectorHttpOutput(b, framework.Forwarder.Spec.Outputs[0])
+				}),
+			).To(BeNil())
+			type K8sAuditLogArray struct {
+				Items []types.K8sAuditLog `json:"items,omitempty"`
```

Comment:
These types are ones we redefined in our repo but are these inputs not really audit logs from the core library?  It would be better to test those types

Reply:
I tried plugging in Event and EventList from k8s.io/apiserver/pkg/apis/audit into this test, but they lack JSON tags and don't marshal and unmarshal well. I tested the HTTP input with a live audit feed from a kube API server and it worked.

Reply:
What is special about [this](https://github.com/openshift/eventrouter/blob/release-5.7/sinks/eventdata.go#L61) serializer which takes v1.Events and writes them to STDOUT as a json string rep of Event?  If we can not make it work the way we think we should at a minimum log an issue upstream to add the tags

Reply:
+1,  we should use https://github.com/kubernetes/apiserver/blob/master/pkg/apis/audit/v1/types.go which is the official public API.
As far as I can tell /kubernetes/apiserver/blob/master/pkg/apis/audit/types.go is an internal detail of the kube-APIserver, which is just a copy (without tags!) of the public API. I imagine it's a historical remainder or a mistake by someone who was confused about go packages an thought they needed a copy in the same package.